app/src/androidTest/java/com/android/sample/screen/mainscreen/MainScreenTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.android.sample.screen.mainscreen
+
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import com.android.sample.ui.mainscreen.MainScreen
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import com.android.sample.ui.navigation.Screen
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+
+class MainScreenTest {
+
+  private lateinit var navigationActions: NavigationActions
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    // Mock the NavigationActions
+    navigationActions = mock(NavigationActions::class.java)
+
+    // Mock the current route
+    `when`(navigationActions.currentRoute()).thenReturn(Route.MAIN)
+
+    // Set the content of the screen for testing
+    composeTestRule.setContent { MainScreen(navigationActions) }
+  }
+
+  @Test
+  fun testMainScreenDisplaysProfileSection() {
+    // Check that the profile picture is displayed
+    composeTestRule.onNodeWithTag("ProfilePicture").assertIsDisplayed()
+
+    // Check that the welcome text is displayed
+    composeTestRule
+        .onNodeWithTag("WelcomeText")
+        .assertIsDisplayed()
+        .assertTextContains("What's up Micheal?")
```

Comment:
I would suggest not using a hardcoded value here in the test as in the future the UI may change.

Reply:
I removed this test as the name is hardcoded for the moment.